docs/adrs/0076-partial-update-patches.md

```diff
@@ -0,0 +1,149 @@
+---
+title: "0076 Partial Update Patches"
+---
+
+# *0076 Partial Update Patches*
+
+**🔒 User Story:** *[MB-14102](https://dp3.atlassian.net/browse/MB-14102)*
+
+At present, the MilMove back-end does not have developed support for partial updates or deleting fields in its models through patches. [ADR 0066](docs/adrs/0066-use-custom-nullable-types-for-patch-requests.md) introduced custom nullable types for patch requests which addresses this issue at the swagger/handler layer, but services may not be prepared to fully support the rest of this sort of workflow.
```

Comment:
I think the PPM shipment updater allows for partial updates. I think for PPMs we have some dependent fields that the service keeps track of clearing on updates like if an advance is changed and we need to clear the amount.

Is the other issue here that MTOShipment just has too many nullable fields because it is used to support HHG, NTS, and NTSR shipments and that really should be refactored?  How many fields really need to be nullable/deletable could we see a breakdown of how hard that would be to handle on a case by case basis in a merge function or dedicated endpoints? Is it partially that our frontends are sending nulls when maybe we should be clearer about treating omitted keys differently?

Reply:
This is a great point, and I would say that this ADR attempts to introduce this pattern as an option for our endpoints in cases where it would be beneficial. I don't think it's necessary or even helpful to refactor all of our endpoints, but some fields in the MTOShipment that should be nullable aren't currently nullable, and future models might also have this problem.
I also think that regardless of what happens in the frontend, as long as we're using our database models in the request payloads (and convert to them in the handlers), this problem will exist without some sort of big refactor.

Reply:
>I don't think it's necessary or even helpful to refactor all of our endpoints, but some fields in the MTOShipment that should be nullable aren't currently nullable, and future models might also have this problem.

I'd suggest adding this to ADR to make this point clear. Also make sure to state earlier in the ADR that the MTOShipment fields are the narrow focus here. That could help make approval on this ADR easier.